user-operator/src/main/java/io/strimzi/operator/user/model/KafkaUserModel.java

```diff
@@ -672,4 +696,13 @@ public String getSaslJsonConfig() {
         return getSaslJsonConfig(getScramUserName(name), scramSha512Password);
     }
 
+    /**
+     * Retrieves the compiled regular expression pattern used to exclude certain labels from being included in the secret's metadata.
+     * This pattern is applied to each label key, and labels matching this pattern are not included.
+     *
+     * @return the compiled {@link Pattern} representing the label exclusion criteria, or {@code null} if no exclusion pattern is set.
+     */
+    public Pattern getSecretLabelExclusionPattern() {
```

Comment:
Doesn't seem to be used anywhere?

Reply:
It is used in the `testFromCrdTlsExternalUser` test case.

Reply:
Why does the test need to get it from the model? Does it not know what it set in the config? Also, it would probably not require public access, or?

Reply:
```java
@Test
    public void testFromCrdTlsExternalUser()   {
        KafkaUserModel model = KafkaUserModel.fromCrd(ResourceUtils.createKafkaUser(ResourceUtils.NAMESPACE, new KafkaUserTlsExternalClientAuthentication()), UserOperatorConfig.SECRET_PREFIX.defaultValue(), Boolean.parseBoolean(UserOperatorConfig.ACLS_ADMIN_API_SUPPORTED.defaultValue()), UserOperatorConfig.SECRET_LABELS_EXCLUSION_PATTERN.type().parse(UserOperatorConfig.SECRET_LABELS_EXCLUSION_PATTERN.defaultValue()));

        assertThat(model.namespace, is(ResourceUtils.NAMESPACE));
        assertThat(model.name, is(ResourceUtils.NAME));
        assertThat(model.labels, is(Labels.fromMap(ResourceUtils.LABELS)
                        .withStrimziKind(KafkaUser.RESOURCE_KIND)
                        .withKubernetesName(KafkaUserModel.KAFKA_USER_OPERATOR_NAME)
                        .withKubernetesInstance(ResourceUtils.NAME)
                        .withKubernetesPartOf(ResourceUtils.NAME)
                        .withKubernetesManagedBy(KafkaUserModel.KAFKA_USER_OPERATOR_NAME)));
        assertThat(model.authentication.getType(), is(KafkaUserTlsExternalClientAuthentication.TYPE_TLS_EXTERNAL));
        assertThat(model.isTlsExternalUser(), is(true));
        assertThat(model.getUserName(), is("CN=" + ResourceUtils.NAME));
        assertThat(model.generateSecret(), is(nullValue()));
        assertThat(model.getSecretLabelExclusionPattern(), is(nullValue()));
    }
    ```
 In that case there is only model so we can't infer such value from config I guess.   
 
 > Also, it would probably not require public access, or
 
Yes, that's right, protected should be fine.

Reply:
No, this shows that you are testing the wrong thing. You should test the labels of the secret and not some configuration value.

Reply:
Right, I would remove it then.